docs/spec/store/README.md

```diff
@@ -0,0 +1,170 @@
+`Set` and `Delete` calls are executed on the in-memory cache, whereas `Has` calls
+are proxied to the underlying `KVStore`. 
+
+One of the most important calls to a `cachekv.Store` is `Write()`, which ensures
+that key-value pairs are written to the underlying `KVStore` in a deterministic
+and ordered manner by sorting the keys first. The store keeps track of "dirty"
+keys and uses these to determine what keys to sort. In addition, it also keeps
+track of deleted keys and ensures these are also removed from the underlying
+`KVStore`.
+
+The `cachekv.Store` also provides the ability to perform iteration and reverse
+iteration. Iteration is performed through the `cacheMergeIterator` type and uses
+both the dirty cache and underlying `KVStore` to iterate over key-value pairs.
+
+Note, all calls to CRUD and iteration operations on a `cachekv.Store` are thread-safe.
+
+### `gaskv.Store`
+
+The `gaskv.Store` store provides a simple implementation of a `KVStore`.
+Specifically, it just wraps an existing `KVStore`, such as a cache-wrapped
+`iavl.Store`, and incurs configurable gas costs for CRUD operations via
+`ConsumeGas()` calls defined on the `GasMeter` which exists in a `sdk.Context`
+and then proxies the underlying CRUD call to the underlying store. Note, the
+`GasMeter` is reset on each block.
+
+### `cachemulti.Store` & `rootmulti.Store`
+
+The `rootmulti.Store` acts as an abstraction around a series of stores. Namely,
+it implements the `CommitMultiStore` an `Queryable` interfaces. Through the
+`rootmulti.Store`, an SDK module can request access to a `KVStore` to perform
+state CRUD operations and queries by holding access to a unique `KVStoreKey`.
+
+The `rootmulti.Store` ensures these queries and state operations are performed
+through cached-wrapped instances of `cachekv.Store` which is described above. The
+`rootmulti.Store` implementation is also responsible for committing all accumulated state from each `KVStore` to disk and returning an application state Merkle root.
+
+Queries can be performed to return state data along with associated state
+commitment proofs for both previous heights/versions and the current state root.
+Queries are routed based on store name, i.e. a module, along with other parameters
+which are defined in `abci.RequestQuery`.
+
+The `rootmulti.Store` also provides primitives for pruning data at a given
+height/version from state storage. When a height is committed, the `rootmulti.Store`
+will determine if other previous heights should be considered for removal based
+on the operator's pruning settings defined by `PruningOptions`, which defines
+how many recent versions to keep on disk and the interval at which to remove
+"staged" pruned heights from disk. During each interval, the staged heights are
+removed from each `KVStore`. Note, it is up to the underlying `KVStore`
+implementation to determine how pruning is actually performed.
+
+It is important to note that the `rootmulti.Store` considers each `KVStore` as a
```

Comment:
seems like the explanation of what the rootmultistore is here, would be helpful to move higher before we start talking about what it does

Reply:
I'm not following? This section is about the root multi-store.